### PR TITLE
fix(core) rpc handling of cjson.null

### LINKED ISSUE #8610: Using getBody and sending a JSON payload with null values causes error
When using `kong.request.getBody` and sending a JSON payload including a null value, like
```
{ "key": null }
```
causes an error
```
2022/03/02 10:26:20 [error] 134#0: *3236 [kong] init.lua:297 [smart-logic] /usr/local/share/lua/5.1/MessagePack.lua:65: pack 'userdata' is unimplemented"
```

As a workaround I had to use the `kong.request.getRawBody()` and `JSON.parse`, but it's not the ideal solution.

Kong version: 2.7.1
Kong JS PDK version: 0.4.4 (I cannot upgrade to the latest one because of this: https://github.com/Kong/kong-js-pdk/issues/151 )

### PR DESCRIPTION
**Pull request by Suika-Kong**
RPC(msgpack) did not recongnize cjson.null(userdata NULL).
Patch to make it compatible with cjson.

Fix #8610

### CODE REVIEW

**Review by kikito — DISMISSED** [review]
This change looks ok to me, but needs two more changes:
* One test
* An entry in the changelog, on the Fixes section.

File: kong/runloop/plugin_servers/mp_rpc.lua [hunk 1]
```diff
@@ -122,6 +133,9 @@ local function call_pdk_method(cmd, args)
 
   local saved = Rpc.save_for_later[coroutine.running()]
   if saved and saved.plugin_name then
+    if ngx.ctx ~= nil then
```
Comment (fffonion) [inline kong/runloop/plugin_servers/mp_rpc.lua:136]: looks like this one is from a different PR, let's remove it from here.

File: kong/runloop/plugin_servers/mp_rpc.lua [hunk 2]
```diff
@@ -122,6 +133,9 @@ local function call_pdk_method(cmd, args)
...
+      ngx.ctx.KONG_PHASE = saved.ngx_ctx.KONG_PHASE
+    end
```
Comment (Suika-Kong) [inline kong/runloop/plugin_servers/mp_rpc.lua:138]:
```suggestion
```

File: kong/runloop/plugin_servers/mp_rpc.lua [hunk 3]
```diff
@@ -1,6 +1,17 @@
 local kong_global = require "kong.global"
 local cjson = require "cjson.safe"
-local msgpack = require "MessagePack"
+local msgpack do
+  msgpack = require "MessagePack"
+  local nil_pack = msgpack.packers["nil"]
+  -- let msgpack encode cjson.null
+  function msgpack.packers.userdata (buffer, userdata)
+    if userdata == cjson.null then
```
Comment (fffonion) [inline kong/runloop/plugin_servers/mp_rpc.lua:8]:
```suggestion
    if userdata == cjson.null or userdata == ngx.null then
```

File: kong/runloop/plugin_servers/mp_rpc.lua [hunk 4]
```diff
@@ -1,6 +1,17 @@
 local kong_global = require "kong.global"
 local cjson = require "cjson.safe"
-local msgpack = require "MessagePack"
+local msgpack do
+  msgpack = require "MessagePack"
+  local nil_pack = msgpack.packers["nil"]
+  -- let msgpack encode cjson.null
```
Comment (fffonion) [inline kong/runloop/plugin_servers/mp_rpc.lua:6]:
```suggestion
  -- let msgpack encode cjson.null and ngx.null
```

**Review by bungle — APPROVED** [review]

**Review by kikito — APPROVED** [review]

**Pull request merged**